packages/react/src/components/ComboBox/ComboBox.tsx

```diff
@@ -746,7 +750,10 @@ const ComboBox = forwardRef(
         setHighlightedIndex(indexToHighlight(normalizedInput));
       },
       onSelectedItemChange({ selectedItem }) {
-        onChange({ selectedItem });
+        // only call onChange if new selection is updated from previous
+        if (JSON.stringify(selectedItem) !== JSON.stringify(selectedItemProp)) {
```

Comment:
FilterableMultiSelect and MultiSelect use `isEqual` from `react-fast-compare` for this. It's faster and handles a bunch of edge cases that would break or be wrong with JSON.stringify.

Reply:
Nice, good info, updated!